Triggers refactor

matis755 opened the pull request.

matis755 committed:
Add new version of TriggerIn

matis755 committed:
Add tests for TriggerIn

matis755 committed:
Add new version of TriggerOut

matis755 committed:
Add tests for TriggerOut

matis755 committed:
Add new version of TriggerInOut

matis755 committed:
Add tests for TriggerInOut

matis755 committed:
Code cleanup

r4jmond commented on src/trigger_out.h:
```diff
@@ -3,17 +3,19 @@
     class TriggerOut {
     public:
-        bool IsNotifier() const;
+        sm::SMStateType GetSMState() const;
```
Change to GetNotifier

matis755 replied:
Done

r4jmond commented on src/trigger_in.h:
```diff
@@ -3,17 +3,22 @@
 
     private:
-        bool trigger;
+        bool isTrigger = false;
+        sm::SMStateType trigger;
```
Please create another type for trigger containing sm::SMStateType and isTrigger. We should only have field 'trigger' according to requirements.

matis755 replied:
Done

r4jmond commented on tests/test_trigger_inout.cpp:
```diff
@@ -0,0 +1,20 @@
+
+    // SM part
+    EXPECT_TRUE(mySM.triggerInOut.IsTrigger());
+    mySM.triggerInOut.SetTransitionResult(ara::sm::ErrorType::kSuccess, ara::sm::SMStateType::On);
+    mySM.triggerInOut.DiscardTrigger();
```
SM part? do you mean that this should be executed from SM?
If so, move it to SM, we will clean SM later

matis755 replied:
Done

r4jmond commented on tests/test_trigger_out.cpp:
```diff
@@ -0,0 +1,7 @@
+
+/** @brief Fills [SWS_SM_00020] */
+TEST_F(smTests, testInternalStatePropagation) {
+    mySM.triggerOut.SetNotifier(ara::sm::SMStateType::On);
+    EXPECT_EQ(mySM.triggerOut.GetSMState(), ara::sm::SMStateType::On);
```
This test shall set SM to ON. Line 5 should be moved to SM

matis755 replied:
Done

r4jmond commented on tests/test_trigger_in.cpp:
```diff
@@ -0,0 +1,13 @@
+    EXPECT_EQ(mySM.triggerIn.GetDesiredState(), ara::sm::SMStateType::Off);
+    EXPECT_TRUE(mySM.triggerIn.IsTrigger());
+
+    mySM.triggerIn.DiscardTrigger();
```
Move to SM

matis755 replied:
Done

r4jmond commented on src/trigger_in_out.h:
```diff
@@ -3,22 +3,26 @@
+        sm::TriggerInOutNotifierType GetTransitionResult() const;
+        void SetSMState(sm::SMStateType);
 
+        void SetTransitionResult(sm::ErrorType, sm::SMStateType);
```
pls name these parameters

matis755 replied:
Done

r4jmond commented on src/trigger_in_out.h:
```diff
@@ -3,22 +3,26 @@
-        bool IsNotifier() const;
-
-        void SetNotifier(bool newNotifier);
+        sm::TriggerInOutNotifierType GetTransitionResult() const;
```
Rename to GetNotifier

matis755 replied:
Done

r4jmond commented on src/trigger_in_out.h:
```diff
@@ -3,22 +3,26 @@
-
-        void SetNotifier(bool newNotifier);
+        sm::TriggerInOutNotifierType GetTransitionResult() const;
+        void SetSMState(sm::SMStateType);
```
Rename to SetTrigger

matis755 replied:
Done

r4jmond commented on src/trigger_in_out.h:
```diff
@@ -3,22 +3,26 @@
+        sm::TriggerInOutNotifierType notifier;
+        sm::SMStateType trigger;
+
+        sm::ErrorType transitionResult;
```
Delete this field, unused

matis755 replied:
Done

matis755 committed:
Fix review issues:
- add new data type for TriggerIn
- remove unused parameters
- move SM test part to SM

r4jmond commented on src/sm_types.h:
```diff
@@ -130,5 +130,27 @@ namespace ara::sm {
+    /**
+    * @brief Fills [SWS_SM_91007]
+    * @details Type used for triggering SM state change */
+    class TriggerInMessageType {
```
maybe just TriggerType? Can be used both for TriggerIn and TriggerInOut

matis755 replied:
Done

matis755 committed:
Change sleep mechanism

r4jmond commented on src/trigger_in_out.h:
```diff
@@ -3,22 +3,25 @@
-        bool notifier;
-        bool trigger;
+        sm::TriggerInOutNotifierType notifier;
+        sm::SMStateType trigger;
+
+        bool isTrigger = false;
```
use TriggerType here

matis755 replied:
Done

r4jmond reviewed (commented):
Change TriggerInMessageType to TriggerType and use it in both TriggerIn and TriggerInOut

r4jmond commented on tests/test_trigger_out.cpp:
```diff
@@ -0,0 +1,9 @@
+
+/** @brief Fills [SWS_SM_00020] */
+TEST_F(smTests, testInternalStatePropagation) {
+    // let changes to be applied
```
What changes? You didn't call SMSetState(on)

matis755 replied:
It's no logic implemented yet in SM. Now, SM sets TriggerOut notifier without any conditions

r4jmond reviewed (commented):
please use TriggerType for both TriggerIn and TriggerInOut

matis755 committed:
Used common type for both TriggerIn and TriggerInOut

r4jmond commented on src/trigger_in_out.h:
```diff
@@ -3,22 +3,23 @@
+        sm::TriggerInOutNotifierType GetNotifier() const;
+        void SetTrigger(sm::SMStateType);
 
+        void SetTransitionResult(sm::ErrorType result, sm::SMStateType currentSMState);
```
change to SetNotifier

matis755 replied:
Done

matis755 committed:
Changed method name

r4jmond committed:
Resolved conflicts, added missing features
